code/game/machinery/doors/table.dm

```diff
@@ -0,0 +1,293 @@
+			return 0
+		else
+			visible_message("<span class='warning'>[src] breaks down!</span>")
+			dismantle()
+			return 1
+	return 1
+
+/obj/machinery/door/table/proc/update_adjacent()
+	for(var/direction in cardinal)
+		var/obj/structure/table/T = locate(/obj/structure/table, get_step(src, direction))
+		if(T)
+			T.update_icon()
+
+/obj/machinery/door/table/door_animate(animation) // no spritework for it
+	return
+
+/obj/machinery/door/table/attack_ai(mob/user as mob) //those aren't really machinery, they're just big fucking slabs of a mineral
+	if(isAI(user)) //so the AI can't open it
+		return
+	else if(isrobot(user) && get_dist(user,src) <= 1) //but robots can, not remotely though
+		return TryToSwitchState(user) //also >nesting if statements
+
+/obj/machinery/door/table/attack_paw(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/machinery/door/table/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/machinery/door/table/proc/TryToSwitchState(mob/user as mob)
+	if(operating)
+		return
+
+	if(!user.restrained() && (user.size > SIZE_TINY))
+		add_fingerprint(user)
+		SwitchState()
+	return
+
+/obj/machinery/door/table/proc/SwitchState()
+	if(!density)
+		return close()
+	else
+		return open()
+
+/obj/machinery/door/table/proc/remove_electronics()
+	if (electronics)
+		electronics.forceMove(loc)
+		electronics = null
+	req_access = list()
+	req_one_access = list()
+
+/obj/machinery/door/table/proc/dismantle()
+	remove_electronics()
+	if(sheet_type)
+		new sheet_type(loc,2)
+	qdel(src)
+
+/obj/machinery/door/table/open()
+	playsound(src, soundeffect, 100, 1)
+	return ..()
+
+/obj/machinery/door/table/close()
+	playsound(src, soundeffect, 100, 1)
+	. = ..()
+	set_opacity(0) //always seethru
+
+/obj/machinery/door/table/attackby(obj/item/W as obj, mob/user as mob, params)
+
+	if (!electronics)
+		if(W.is_wrench(user))
+			to_chat(user, "<span class='notice'>Now disassembling [src]...</span>")
+			W.playtoolsound(src, 50)
+			if(do_after(user, src,50))
+				dismantle()
+			return
+
+		if(panel_open && istype(W,/obj/item/weapon/circuitboard/airlock))
+			if(W.icon_state == "door_electronics_smoked")
+				to_chat(user, "<span class='warning'>Repair \the [W] before putting it in!</span>")
+			if(user.drop_item(W,src))
+				electronics = W
+				if(electronics.conf_access?.len)
+					if(electronics.one_access)
+						req_one_access = electronics.conf_access
+					else
+						req_access = electronics.conf_access
+				electronics.installed = TRUE
+				playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
+				to_chat(user, "<span class='notice'>You add [electronics] to [src].</span>")
+			return
+
+	else
+		if(W.is_wrench(user))
+			to_chat(user, "<span class='warning'>Remove [electronics] first!</span>")
+			return
+
+		if(panel_open && iscrowbar(W))
+			user.visible_message("[user] is removing [electronics] from [src].", "You start to remove \the [electronics] from [src].")
```

Comment:
I would wrap this in `<span class='notice'></span>`

Reply:
done